packages/excubiae/contracts/extensions/SemaphoreExcubia.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import {Excubia} from "../Excubia.sol";
+import {ISemaphore} from "@semaphore-protocol/contracts/interfaces/ISemaphore.sol";
+
+/// @title Semaphore Excubia Contract
+/// @notice This contract extends the Excubia contract to integrate with the Semaphore protocol.
+/// It verifies the passerby Semaphore group membership proofs to grant access through the gate.
+/// @dev To allow only specific Semaphore identities from a group, the contract stores the specific group identifier.
+/// To avoid identities from passing twice, nullifiers are stored upon successful verification of the proofs.
+contract SemaphoreExcubia is Excubia {
+    /// @notice The Semaphore contract interface.
+    ISemaphore public immutable SEMAPHORE;
+    /// @notice The specific group identifier that proofs must match to pass the gate.
+    /// @dev Used as a `scope` to ensure consistency during proof membership verification.
+    uint256 public immutable GROUP_ID;
+
+    /// @notice Mapping to track which nullifiers have been used to avoid passing the
+    /// gate twice using the same Semaphore identity.
+    /// @dev The nullifier is derived from the hash of the secret and group identifier,
+    /// ensuring that the same identity cannot be registered twice for the same group.
+    mapping(uint256 => bool) public passedNullifiers;
+
+    /// @notice Error thrown when the group identifier does not match the expected one.
+    error InvalidGroup();
+
+    /// @notice Error thrown when the proof is invalid.
+    error InvalidProof();
+
+    /// @notice Error thrown when the proof scope does not match the expected group identifier.
+    error UnexpectedScope();
```

Comment:
InvalidScope? just cause every other error starts with invalid

Reply:
This naming convention follows the other Excubia. When there's something fixed in the contract (e.g., GROUP_ID in this case), we say it's "unexpected" because passersby can give valid scopes, but this is one we weren't expecting (when different from the fixed one, ie. GROUP_ID).

Let me know if it makes sense!

Reply:
makes sense, pretty neat